test/Microsoft.Extensions.Logging.Test/LoggerTest.cs

```diff
@@ -101,6 +101,22 @@ public void Log_AggregatesExceptionsFromMultipleLoggers()
             Assert.Equal("provider2.Test-Error occurred while logging data.", exceptions[1].Message);
         }
 
+        [Fact]
+        public void LoggerCanGetProviderAfterItIsCreated()
+        {
+            // Arrange
+            var store = new List<string>();
+            var loggerFactory = new LoggerFactory();
+            var logger = loggerFactory.CreateLogger("Test");
+            loggerFactory.AddProvider(new CustomLoggerProvider("provider1", ThrowExceptionAt.None, store));
+
+            // Act
+            logger.LogInformation("Hello");
```

Comment:
Is this fix related to logs not being visible from Kestrel? Have you tested this fix with an end-to-end application?